util/include/crypto/openssl/crypto.hpp

```diff
@@ -110,6 +127,67 @@ class AsymmetricPublicKey {
   AsymmetricPublicKey() {}
 };
 
+// Wrapper class for OpenSSL Crypto's EVP_PKEY objects implementing
+// AsymmetricPrivateKey.
+class EVPPKEYPrivateKey : public AsymmetricPrivateKey {
```

Comment:
This class is needed if we want to use existing code to generate ecdsa keys.
We still need to adapt the interface to support PEM files, which are used by 
 ClientTlsKeyExchangeHandler::exchangeTlsKeys.